flow/diff_context.h

```diff
@@ -0,0 +1,235 @@
+#ifndef FLUTTER_FLOW_DIFF_CONTEXT_H_
+#define FLUTTER_FLOW_DIFF_CONTEXT_H_
+
+#include <map>
+#include <vector>
+#include "flutter/fml/logging.h"
+#include "flutter/fml/macros.h"
+#include "third_party/skia/include/core/SkMatrix.h"
+#include "third_party/skia/include/core/SkRect.h"
+
+namespace flutter {
+
+#ifndef NDEBUG
+#define FLUTTER_ENABLE_DIFF_CONTEXT
+#endif
+
+#ifdef FLUTTER_ENABLE_DIFF_CONTEXT
+
+class Layer;
+
+struct Damage {
+  // This is the damage between current and previous frame;
+  // If embedder supports partial update, this is the region that needs to be
+  // swapped.
+  SkIRect surface_damage;
+
+  // Reflects actual change to target framebuffer; This is surface_damage +
```

Comment:
If surface_damage represents the delta between current and prev frame, is it not to the frame buffer? I'm a little confused by the phrasing "Reflects actual change to target framebuffer". What does it mean for the damage to be preciously accumulated to the target frame-buffer? Does it not get reset between each frame?

Reply:
The naming will change (see comment from @flar below), I'll also add link EGL_KHR_partial_update spec for further explanation.